homeassistant/components/reolink/camera.py

```diff
@@ -17,69 +23,130 @@
+class ReolinkCameraEntityDescription(
+    CameraEntityDescription,
+):
+    """A class that describes camera entities for a camera channel."""
+
+    stream: str
+    supported: Callable[[Host, int], bool] = lambda api, ch: True
+
+
+CAMERA_ENTITIES = (
+    ReolinkCameraEntityDescription(
+        key="sub",
+        stream="sub",
+        translation_key="sub",
+    ),
+    ReolinkCameraEntityDescription(
+        key="main",
+        stream="main",
+        translation_key="main",
+        entity_registry_enabled_default=False,
+    ),
+    ReolinkCameraEntityDescription(
+        key="snapshots_sub",
+        stream="snapshots_sub",
+        translation_key="snapshots_sub",
+        entity_registry_enabled_default=False,
+    ),
+    ReolinkCameraEntityDescription(
+        key="snapshots",
+        stream="snapshots_main",
+        translation_key="snapshots_main",
+        entity_registry_enabled_default=False,
+    ),
+    ReolinkCameraEntityDescription(
+        key="ext",
+        stream="ext",
+        translation_key="ext",
+        supported=lambda api, ch: api.protocol in ["rtmp", "flv"],
+        entity_registry_enabled_default=False,
+    ),
+    ReolinkCameraEntityDescription(
+        key="autotrack_sub",
+        stream="autotrack_sub",
+        translation_key="autotrack_sub",
+        supported=lambda api, ch: api.supported(ch, "autotrack_stream"),
+    ),
+    ReolinkCameraEntityDescription(
+        key="autotrack_snapshots_sub",
+        stream="autotrack_snapshots_sub",
+        translation_key="autotrack_snapshots_sub",
+        supported=lambda api, ch: api.supported(ch, "autotrack_stream"),
+        entity_registry_enabled_default=False,
+    ),
+    ReolinkCameraEntityDescription(
+        key="autotrack_snapshots_main",
+        stream="autotrack_snapshots_main",
+        translation_key="autotrack_snapshots_main",
+        supported=lambda api, ch: api.supported(ch, "autotrack_stream"),
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up a Reolink IP Camera."""
     reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
-    host = reolink_data.host
 
-    cameras = []
-    for channel in host.api.stream_channels:
-        streams = ["sub", "main", "snapshots_sub", "snapshots_main"]
-        if host.api.protocol in ["rtmp", "flv"]:
-            streams.append("ext")
-
-        if host.api.supported(channel, "autotrack_stream"):
-            streams.extend(
-                ["autotrack_sub", "autotrack_snapshots_sub", "autotrack_snapshots_main"]
+    entities: list[ReolinkCamera] = [
+        ReolinkCamera(reolink_data, channel, entity_description)
+        for entity_description in CAMERA_ENTITIES
+        for channel in reolink_data.host.api.stream_channels
+        if entity_description.supported(reolink_data.host.api, channel)
```

Comment:
My impression is the best practice is to not use inline for loops when the if statements span multiple lines.

Reply:
I expended the for loop, thanks for the feedback :)